api/v1alpha1/types_etcd.go

```diff
@@ -485,3 +567,48 @@ func (e *Etcd) GetRoleName() string {
 func (e *Etcd) GetRoleBindingName() string {
 	return fmt.Sprintf("%s:etcd:%s", GroupVersion.Group, e.Name)
 }
+
+// IsBackupStoreEnabled returns true if backup store has been enabled for this etcd, else returns false.
+func (e *Etcd) IsBackupStoreEnabled() bool {
+	return e.Spec.Backup.Store != nil
+}
+
+// IsMarkedForDeletion returns true if a deletion timestamp has been set and false otherwise.
+func (e *Etcd) IsMarkedForDeletion() bool {
+	return !e.DeletionTimestamp.IsZero()
+}
+
+// GetSuspendEtcdSpecReconcileAnnotationKey gets the annotation key set on an etcd resource signalling the intent
+// to suspend spec reconciliation for this etcd resource. If no annotation is set then it will return nil.
+func (e *Etcd) GetSuspendEtcdSpecReconcileAnnotationKey() *string {
+	if metav1.HasAnnotation(e.ObjectMeta, SuspendEtcdSpecReconcileAnnotation) {
+		return pointer.String(SuspendEtcdSpecReconcileAnnotation)
+	}
+	if metav1.HasAnnotation(e.ObjectMeta, IgnoreReconciliationAnnotation) {
+		return pointer.String(IgnoreReconciliationAnnotation)
+	}
+	return nil
+}
+
+// IsReconciliationSuspended returns true if the etcd resource has the annotation set to suspend spec reconciliation,
+// else returns false.
+func (e *Etcd) IsReconciliationSuspended() bool {
+	suspendReconcileAnnotKey := e.GetSuspendEtcdSpecReconcileAnnotationKey()
+	return suspendReconcileAnnotKey != nil && metav1.HasAnnotation(e.ObjectMeta, *suspendReconcileAnnotKey)
+}
```

Comment:
`IsReconciliationSuspended` is a convenience method, which can be used by any consumer to know whether reconciliation of the Etcd is active or suspended. Especially since there are two annotations currently to achieve suspension of reconciliation, this method provides an easy way to check this, rather than users having to check for both the annotations by themselves.

Reply:
Thanks for the explanation.